modules/apps/portal-search/portal-search-ranking-web/src/main/resources/META-INF/resources/js/components/list/SearchBar.es.js

```diff
@@ -19,21 +19,21 @@ class SearchBar extends Component {
 		onClickPin: PropTypes.func,
 		onSelectAll: PropTypes.func.isRequired,
 		onSelectClear: PropTypes.func.isRequired,
+		onUpdateSearchBarTerm: PropTypes.func,
 		resultIds: PropTypes.arrayOf(String),
-		selectedIds: PropTypes.arrayOf(String),
-		showAddResultButton: PropTypes.bool
+		searchBarTerm: PropTypes.string,
+		selectedIds: PropTypes.arrayOf(String)
 	};
 
 	static defaultProps = {
 		resultIds: [],
-		selectedIds: [],
-		showAddResultButton: true
+		selectedIds: []
 	};
 
 	selectAllCheckbox = React.createRef();
 
 	state = {
-		resultSearch: ''
+		resultSearch: this.props.searchBarTerm
```

Comment:
It's an anti-pattern to copy props into state unless you're expecting the prop to never change (i.e. `initialValue`). But since `searchBarTerm` is a state on the `ResultsRankingForm`, if that component ever updates its state, that state will never be updated in `SearchBar` since the state was only copied when `SearchBar` was first created.

Instead we should just be using the prop `searchBarTerm` and using a callback from to update the value.